cargo-miri/bin.rs

```diff
@@ -1181,6 +1190,13 @@ fn main() {
             let binary = Path::new(arg);
             if binary.exists() {
                 assert!(!arg.starts_with("--")); // not a flag
+                if let Some(rustc) = std::env::var_os("RUSTC") {
```

Comment:
If we need to keep this, then it should be the very first thing in `Some(arg) =>`, with a comment like
```
// When both RUSTC_WRAPPER and RUSTC are set, cargo ends up invoking
// '$RUSTC_WRAPPER $RUSTC <args>'. So our `phase_rustc` check above might be ineffective, and we
// need another check here.
```